Library/Homebrew/utils/versions.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module Versions
+  module_function
+
+  def current_formula_version(formula_name)
+    Formula[formula_name].version.to_s.to_f
```

Comment:
This will truncate the formula version if it has any non-numeric components, or a three-part semver version. Suggest you work directly with the `Version` object when possible.

```console
% brew irb
==> Interactive Homebrew Shell
Example commands available with: brew irb --examples

irb(main):001:0> "x264".f.version.to_s
=> "r2999"
irb(main):002:0> "x264".f.version.to_s.to_f
=> 0.0
irb(main):003:0> "curl".f.version.to_s
=> "7.71.1"
irb(main):004:0> "curl".f.version.to_s.to_f
=> 7.71
```

Reply:
Agreed.